doc/neps/nep-0050-scalar-promotion.rst

```diff
@@ -0,0 +1,743 @@
+   :ref:`NEP 42 <NEP42>`.
+   Currently, this leads to a dual implementation of a new and an old (value
+   sensitive) system.  Fixing this will greatly simplify the internal logic
+   and make results more consistent.
+
+We believe that the proposal of "weak" Python scalars will help users by
+providing a clear mental model for which datatype an operation will
+result in.
+This model fits well with the preservation of array precisions that NumPy
+currently often follows, and also uses for in-place operations::
+
+    arr += value
+
+Preserves precision as long as "kind" boundaries are not crossed (otherwise
+an error is raised).
+
+While some users will potentially miss the value inspecting behavior even for
+those cases where it seems useful, it quickly leads to surprises.  This may be
+expected::
+
+    np.array([100], dtype=np.uint8) + 1000 == np.array([1100], dtype=np.uint16)
+
+But the following will then be a surprise::
+
+    np.array([100], dtype=np.uint8) + 200 == np.array([44], dtype=np.uint8)
+
+Considering that the proposal aligns with the behavior of in-place operands
+and avoids the surprising switch in behavior that only sometimes avoids
+overflow in the result,
+we believe that the proposal follows the "principle of least surprise".
+
+
+Usage and Impact
+================
+
+This NEP is expected to be implemented with **no** transition period that warns
+for all changes.  Such a transition period would create many (often harmless)
+warnings which would be difficult to silence.
+We expect that most users will benefit long term from the clearer promotion
+rules and that few are directly (negatively) impacted by the change.
+However, certain usage patterns may lead to problematic changes, these are
+detailed in the backwards compatibility section.
+
+The solution to this will be an *optional* warning mode capable of notifying
+users of potential changes in behavior.
+This mode is expected to generate many harmless warnings, but provide a way
+to systematically vet code and track down changes if problems are observed.
+
+
+Impact on ``can_cast``
+----------------------
+
+`can_cast` will never inspect the value anymore.  So that the following results
+are expected to change from ``True`` to ``False``::
+
+  np.can_cast(np.int64(100), np.uint8)
+  np.can_cast(np.array(100, dtype=np.int64), np.uint8)
+  np.can_cast(100, np.uint8)
+
+We expect that the impact of this change will be small compared to that of
+the following changes.
+
+.. note::
+
+    The last example where the input is a Python scalar _may_ be preserved
+    since ``100`` can be represented by a ``uint8``.
+
+
+Impact on operators and functions involving NumPy arrays or scalars
+-------------------------------------------------------------------
+
+The main impact on operations not involving Python scalars (``float``, ``int``,
+``complex``) will be that 0-D arrays and NumPy scalars will never behave
+value-sensitive.
```

Comment:
```suggestion
value-sensitively.
```

Although, this phrasing is slightly odd; can reword perhaps to something like "that operations on 0-D arrays and NumPy scalars will never depend on their values"